marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/Api.hs

```diff
@@ -141,6 +143,10 @@ data Metadata
 instance ToJSON Metadata where
   toJSON = metadataValueToJsonNoSchema . toCardanoMetadata
 
+-- Should probably use metadataValueFromJsonNoSchema from Cardano.Api instead
+instance FromJSON Metadata where
+  parseJSON v = maybe (A.typeMismatch "Metadata" v) pure $ fromJSONEncodedMetadata v
```

Comment:
Check with colleagues how to proceed here.

Reply:
@jhbertra This bit of code is interesting.

`Metadata` seems like it should have an apparent JSON structure but there seems to be two structures after looking in `Cardano.Api`. We're using the seemingly less verbose one referred to as the "no schema" one. We use a procedure `metadataValueToJsonNoSchema` in the `ToJSON` instance. There is a procedure `metadataValueFromJsonNoSchema` in `Cardano.Api` which looks perfect for a `FromJSON` instance here but it isn't exported. I don't know how big of a ceremony it would be to have it exported since it's in another repo (assuming that's even the right approach).

The current solution using `fromJSONEncodedMetadata` is apparently explicitly documented as being inappropriate for a `FromJSON` instance implementation.

Thoughts?